### PR TITLE
Plane support for Frsky passthrough protocol

### PR DESCRIPTION
**Pull request by floaledm**
Based on the work in #4762 and #4764, so some of the commits are in common...

Initial Plane support, tested on bench, but not in flight.

### CODE REVIEW

**Comment by floaledm** [comment]
Looks like there's an issue with one of the sitl tests. Let me know what I can do to remedy this...

File: ArduPlane/is_flying.cpp [hunk 1]
```diff
@@ -155,6 +155,9 @@ void Plane::update_is_flying_5Hz(void)
     }
     previous_is_flying = new_is_flying;
     adsb.set_is_flying(new_is_flying);
+#if FRSKY_TELEM_ENABLED == ENABLED
+    frsky_telemetry.set_is_flying(new_is_flying);
```
Comment (magicrub) [inline ArduPlane/is_flying.cpp]: I see you copied the ADSB example on the line above, sensible thing to do. However, that method is wrong and needs to be fixed. It should use the probability and read as:
frsky_telemetry.set_is_flying(is_flying());

Comment (floaledm) [inline ArduPlane/is_flying.cpp]: Thanks! Just changed it to frsky_telemetry.set_is_flying(is_flying());

**Comment by tridge** [comment]
for testing changes like these, have you and Tom tried testing with SITL so you can fly any vehicle type and test with a real receiver and taranis transmitter? Just use the --uartE option to SITL to point at a FTDI cable connected to your receiver

File: libraries/AP_Frsky_Telem/AP_Frsky_Telem.h [hunk 2]
```diff
@@ -126,6 +127,9 @@ class AP_Frsky_Telem
     // update flight control mode. The control mode is vehicle type specific
     void update_control_mode(uint8_t mode) { _ap.control_mode = mode; }
 
+    // update whether we're flying (used for Plane)
+    void set_is_flying(bool is_flying) { *_ap.value == is_flying ? (*_ap.value | AP_ISFLYING_FLAG) : (*_ap.value & (~AP_ISFLYING_FLAG)); }
```
Comment (floaledm) [inline libraries/AP_Frsky_Telem/AP_Frsky_Telem.h]: Cleared a typo here, now works!

Comment (magicrub) [inline libraries/AP_Frsky_Telem/AP_Frsky_Telem.h]: is there a PR for that?

Comment (floaledm) [inline libraries/AP_Frsky_Telem/AP_Frsky_Telem.h]: I had a typo in this PR which I corrected. This line of code is new to this PR (not in master). Now passes sitl...

**Comment by magicrub** [comment]
applied, thanks! I made one minor change ( https://github.com/ArduPilot/ardupilot/commit/b8d53e1c58cb7a76bf5ef42b02720cfa9f80c8a5 ) which was to correct by bad advice I had given you.

**Comment by floaledm** [comment]
Thanks Tom!

**Comment by magicrub** [comment]
@floaledm now that init() contains the platform type (MAV_TYPE_FIXED_WING) I'd like to see the other inits get squashed into a single init. If you don't think thats possible due to MAV_TYPE limitations then I'd like to at least see the init() functions renamed to init_plane() and init_rover()

**Comment by floaledm** [comment]
@magicrub Good suggestion. I'll issue another PR to address that...

**Comment by floaledm** [comment]
@magicrub Before I get ahead of myself here, since each init() has (for now) a different number of arguments, do you have a suggestion on how to squash them in a single init?

**Comment by magicrub** [comment]
nullptr or NULL or 0. Those values won't be used when they are not populated so it technicially shoudln't natter what they are. However, it's always a good idea to check for null references before using any object so you should throw some error or statusTEXT or something (or fail the init) when that happens.

**Pull request closed without merging**